containers/src/Data/IntMap.hs

```diff
@@ -50,10 +50,24 @@
 --
 -- Operation comments contain the operation time complexity in
 -- the Big-O notation <http://en.wikipedia.org/wiki/Big_O_notation>.
+--
 -- Many operations have a worst-case complexity of \(O(\min(n,W))\).
 -- This means that the operation can become linear in the number of
 -- elements with a maximum of \(W\) -- the number of bits in an 'Int'
--- (32 or 64).
+-- (32 or 64). This peculiar asymptotics is determined by the depth
+-- of Patricia trees:
+--
+-- * even for extremely unbalanced tree the depth cannot be larger than
+--   the number of elements \(n\),
+-- * each level of a Patricia tree determines at least one more bit
+--   shared by all subelements, so there could not be more
+--   than \(W\) levels.
+--
+-- If all \(n\) keys in the tree are less than \(N\),
```

Comment:
Fair enough, I'm only seeing if the situation can be generalized to give the reader more information. But it might be good enough to address only the typical scenario.